Adding tensor names for TFLite Java

kmader opened the pull request:
Add the commands to get the input and output names for TFLite java
- Closes #27360

kmader committed:
Adding tensor names for TFLite Java

Add the commands to get the input and output names for TFLite java
- Closes #27360

kmader committed:
Update NativeInterpreterWrapper.java

adding `getInputTensorNames` and `getOutputTensorNames` to the private wrapper

lu-wang-g commented on tensorflow/lite/java/src/main/java/org/tensorflow/lite/NativeInterpreterWrapper.java:
```diff
@@ -268,6 +272,11 @@ int getInputTensorCount() {
   }
 
+  /** Gets the names of the input tensors. */
+  String[] getInputTensorNames() {
```
Better to check if the name exists.

kmader replied:
Sure, I think the `init` method has to be called, but probably not a bad idea to add a check anyways.

kmader replied:
Should I just add this block before?
```
    if (inputTensorNames == null) {
      return new String[]
    }
```

lu-wang-g replied:
Either way looks good to me. inputTensorNames=new String[] {} seems better. What do you think?

kmader replied:
Done!

kmader replied:
@lu-wang-g done!

kmader committed:
Update NativeInterpreterWrapper.java

incorporating feedback from @lu-wang-g and setting tensor names in case they are empty

jdduke commented:
We should probably just add the name as a field on the Tensor class in Tensor.java, letting you do something like:
```
getInputTensor(0).name()
```

jdduke commented:
The merge was a bit premature, expect a rollback shortly.

kmader commented:
@jdduke that makes sense, but I saw there was more infrastructure already in place for `getInputTensorNames` (which the python tflite interpreter also has), so maybe it should have the names in the `Tensor` and `Interpreter`.

jdduke commented:
Yeah, we're trying to streamline the Interpreter interface, particularly for properties that are Tensor-specific.
